### ISSUE TITLE
Missing PyErr_Clear() call causes a SystemError

### ISSUE THREAD

**Issue by warsaw** [PyTables/PyTables#486]
I'm in the process of updating Ubuntu to PyTables 3.2.1.  During the run of the test suite in this environment, we hit a test error:

https://bugs.launchpad.net/ubuntu/+source/pytables/+bug/1478146

As it turns out, `__setitem__()` has a bug where if `PyArray_SETITEM()` returns non-zero, there will already be an exception set, but the Cython code tries to raise a `TypeError` on top of that.  Python doesn't like that. :)

`tableextension.pyx` needs to clear the exception set by `PyArray_SETITEM()` before it raises the `TypeError`.  This patch works for us:

```
--- a/tables/tableextension.pyx
+++ b/tables/tableextension.pyx
@@ -40,7 +40,7 @@
 
 # numpy functions & objects
 from hdf5extension cimport Leaf
-from cpython cimport PY_MAJOR_VERSION
+from cpython cimport PY_MAJOR_VERSION, PyErr_Clear
 from libc.stdio cimport snprintf
 from libc.stdlib cimport malloc, free
 from libc.string cimport memcpy, strdup, strcmp, strlen
@@ -1596,6 +1596,7 @@
       if field.ndim == 1:
         ret = PyArray_SETITEM(field, field.data + offset * self._stride, value)
         if ret < 0:
+          PyErr_Clear()
           raise TypeError
       ##### End of optimization for scalar values
       else:
```

**Comment by andreabedini** [comment]
Thanks for the detailed report @warsaw!

**Comment by avalentino** [comment]
@andreabedini if it is OK for you I would like to apply the suggested patch because it actually fixes an issue with Python 3.5.
We can clean the code in a second time of course.

**Comment by andreabedini** [comment]
Thanks for keeping my feet on the ground @avalentino :) Fixed in fb9b917142996e7dce427af6b46e3e64b36d8107.
